crates/typst/src/foundations/calc.rs

```diff
@@ -908,6 +911,63 @@ pub fn quo(
     Ok(floor(dividend.apply2(divisor.v, Div::div, Div::div)))
 }
 
+/// Calculates the euclidean norm of a sequence of values.
+///
+/// ```example
+/// #calc.norm(1, 2, -3, 0.5)
+/// #calc.norm(1in, 2cm)
+/// #calc.norm(3em, 4em)
+/// ```
+#[func]
+pub fn norm(
+    /// The sequence of values from which to calculate the norm.
+    /// Must not be empty.
```

Comment:
I think this comment can be changed now that empty parameter lists are allowed:

```suggestion
    /// Returns `0.0` if empty.
```